src/CommonProviderImplementation/Helpers.fs

```diff
@@ -74,6 +74,7 @@ type DisposableTypeProviderForNamespaces() as x =
     inherit TypeProviderForNamespaces()
   
     let disposeActions = ResizeArray()
+    let dictLock = obj()
```

Comment:
Book "CLR via C#" has described that there is a bug in .NET framework that if the locking object is used directly it may cause deadlock and for that reason you should use separate locking object.